lib/config/config-validator.js

```diff
@@ -177,6 +178,28 @@ function validateRules(rulesConfig, ruleMapper, source = null) {
     });
 }
 
+/**
+ * Validates a `globals` section of a config file
+ * @param {Object} globalsConfig The `glboals` section
+ * @param {string|null} source The name of the configuration source to report in the event of an error.
+ * @returns {void}
+ */
+function validateGlobals(globalsConfig, source = null) {
+    if (!globalsConfig) {
+        return;
+    }
+
+    Object.keys(globalsConfig)
+        .map(key => globalsConfig[key])
+        .forEach(configuredValue => {
+            try {
+                ConfigOps.normalizeConfigGlobal(configuredValue);
+            } catch (err) {
+                throw new Error(`ESLint configuration in ${source} is invalid:\n${err.message}`);
```

Comment:
Can we improve the UX here by tracking the global key?

For example, if something like `/* global foo: not-a-valid-value */`, could we say something like the below?

```
ESLint configuration of global 'foo' is invalid:\n${err.message}
```